pkg/network/node/networkpolicy.go

```diff
@@ -361,6 +369,22 @@ func (np *networkPolicyPlugin) selectNamespacesInternal(selector labels.Selector
 	return match.matches
 }
 
+func (np *networkPolicyPlugin) selectPodsInternal(namespaceName string, selector labels.Selector) map[string]string {
+	cacheKey := selector.String()
+	match := np.podMatchCache[namespaceName][cacheKey]
+	if match == nil {
+		np.podMatchCache[namespaceName] = make(map[string]*npPodCacheEntry)
```

Comment:
You need to have two checks here, one for if `ns.podMatchCache[namespaceName]` exists, and one for if `ns.podMatchCache[namespaceName][cacheKey]` exists. The way the code is now, if you call `selectPodsInternal("namespace1", selectorA)` and then `selectPodsInternal("namespace1", selectorB)`, then the second call will reinitialize `np.podMatchCache["namespace1"]` with an empty map again and delete the cached results for `selectorA`.

Reply:
I see what your saying and I believe that I corrected it